### ISSUE TITLE
[docker][autoscaler] default nofile ulimit is extremely low

### ISSUE THREAD

**Issue by wuisawesome** [ray-project/ray#13045]
### What is the problem?

When starting a new ray cluster, the ulimit inside the docker container is very low. This is a problem because our recommended `ulimit -n 65535; ray start --head ...` approach runs into a permission denied within docker. 

The proper solution is to do `docker run --ulimit nofile=65535 ...`. The implementation here should mirror  #11953. A reasonable alternative would be to have it as a default in the docker run options. 

*Ray version and other system information (Python version, TensorFlow version, OS):*

### Reproduction (REQUIRED)
Please provide a short code snippet (less than 50 lines if possible) that can be copy-pasted to reproduce the issue. The snippet should have **no external library dependencies** (i.e., use fake or mock data / environments):

If the code snippet cannot be run by itself, the issue will be closed with "needs-repro-script".

- [ ] I have verified my script runs in a clean environment and reproduces the issue.
- [ ] I have verified the issue also occurs with the [latest wheels](https://docs.ray.io/en/master/installation.html).

**Comment by ijrsvt** [comment]
It looks like (by default), Docker sets itself to use a `NOFILE = 1048576` at ([here](https://github.com/moby/moby/blob/46cdcd206c56172b95ba5c77b827a722dab426c5/contrib/init/sysvinit-debian/docker#L96), or [here](https://github.com/moby/moby/blob/d17cf2192c2cda0fa11040b8801a6407636072b8/contrib/init/systemd/docker.service#L15) or in previous versions of conatinerd's SystemD [configuration](https://github.com/containerd/containerd/commit/c691c36614622b205a859ae7e656badb4a553076)). Adding this `--ulimit nofile=65535` would lower the container's max ulimit from 1048576 to 65535. Furthermore if I try to set the value specified for the daemon to a _higher value_ (i.e. `docker run --rm -it -d --ulimit nofile=1048577 rayproject/ray:latest`, I get the following error:
```
docker: Error response from daemon: OCI runtime create failed: container_linux.go:349: starting container process caused "process_linux.go:449: container init caused \"process_linux.go:378: setting rlimits for ready process caused \\\"error setting rlimit type 7: operation not permitted\\\"\"": unknown.
```
### HOWEVER
If I set `containerd` to have `LimitNOFILE=104`, and then restart the `containerd` service in SystemD, I get the following
* Run a container with no `ulimit` flag and run the following inside:
* * `ulimit -n` == `104`
* Run a container with `ulimit nofile=1045876` and run the following inside:
* * `ulimit -n` == `1045876`

TL;DR:
I think we can safely add `ulimit nofile=1048576` as a defaulted docker run option.
